test/test_hydrostatic_free_surface_models.jl

```diff
@@ -99,21 +110,46 @@ end
         for topo in topos
             @testset "Time-stepping HydrostaticFreeSurfaceModels [$arch, $topo]" begin
                 @info "  Testing time-stepping HydrostaticFreeSurfaceModels [$arch, $topo]..."
-                @test time_stepping_hydrostatic_free_surface_model_works(arch, topo, nothing)
+                @test time_stepping_hydrostatic_free_surface_model_works(architecture=arch, topology=topo)
             end
         end
 
         for coriolis in (nothing, FPlane(f=1), BetaPlane(f₀=1, β=0.1))
             @testset "Time-stepping HydrostaticFreeSurfaceModels [$arch, $(typeof(coriolis))]" begin
                 @info "  Testing time-stepping HydrostaticFreeSurfaceModels [$arch, $(typeof(coriolis))]..."
-                @test time_stepping_hydrostatic_free_surface_model_works(arch, topos[1], coriolis)
+                @test time_stepping_hydrostatic_free_surface_model_works(architecture=arch, coriolis=coriolis)
             end
         end
 
-        for advection in (VectorInvariant(), CenteredSecondOrder(), WENO5())
+        for advection in (nothing, VectorInvariant(), CenteredSecondOrder(), WENO5())
             @testset "Time-stepping HydrostaticFreeSurfaceModels [$arch, $(typeof(advection))]" begin
                 @info "  Testing time-stepping HydrostaticFreeSurfaceModels [$arch, $(typeof(advection))]..."
-                @test time_stepping_hydrostatic_free_surface_model_works(arch, topos[1], nothing, advection)
+                @test time_stepping_hydrostatic_free_surface_model_works(architecture=arch, advection=advection)
+            end
+        end
+
+        @testset "Time-stepping HydrostaticFreeSurfaceModels with PrescribedVelocityFields [$arch]" begin
+            @info "  Testing time-stepping HydrostaticFreeSurfaceModels with PrescribedVelocityFields [$arch]"...
+
+                # Non-parameterized functions 
+                u(x, y, z, t) = 1
+                v(x, y, z, t) = exp(z) 
+                w(x, y, z, t) = sin(z) 
```

Comment:
This choice does not satisify the continuity equation.  Do we want one that does, in case problems may arise down the road?

Reply:
heh, that's a fair point. This is really just a unit test that ensures the functions run without throwing an error, so I don't think it matters here.

Reply:
Agreed